pkg/controller/certificates/internal/secretsmanager/secret.go

```diff
@@ -305,7 +317,43 @@ func (s *SecretsManager) setValues(crt *cmapi.Certificate, secret *corev1.Secret
 		secret.Annotations[cmapi.AltNamesAnnotationKey] = strings.Join(x509Cert.DNSNames, ",")
 		secret.Annotations[cmapi.IPSANAnnotationKey] = strings.Join(utilpki.IPAddressesToString(x509Cert.IPAddresses), ",")
 		secret.Annotations[cmapi.URISANAnnotationKey] = strings.Join(utilpki.URLsToString(x509Cert.URIs), ",")
+		secret.Annotations[cmapi.EmailsAnnotationKey] = strings.Join(x509Cert.EmailAddresses, ",")
+		secret.Annotations[cmapi.SubjectOrganizationsAnnotationKey] = strings.Join(x509Cert.Subject.Organization, ",")
+		secret.Annotations[cmapi.SubjectOrganizationalUnitsAnnotationKey] = strings.Join(x509Cert.Subject.OrganizationalUnit, ",")
+		secret.Annotations[cmapi.SubjectCountriesAnnotationKey] = strings.Join(x509Cert.Subject.Country, ",")
+		secret.Annotations[cmapi.SubjectProvincesAnnotationKey] = strings.Join(x509Cert.Subject.Province, ",")
+		secret.Annotations[cmapi.SubjectLocalitiesAnnotationKey] = strings.Join(x509Cert.Subject.Locality, ",")
+<<<<<<< HEAD
+		secret.Annotations[cmapi.SubjectPostalCodesAnnotationKey] = strings.Join(x509Cert.Subject.PostalCode, ",")
+		secret.Annotations[cmapi.SubjectSerialNumberAnnotationKey] = x509Cert.SerialNumber.String()
+		secret.Annotations[cmapi.SubjectStreetAddressesAnnotationKey], err = joinWithEscapeCSV(x509Cert.Subject.StreetAddress)
+
+		if err != nil {
+			return err
+		}
+=======
+		secret.Annotations[cmapi.SubjectStreetAddressesAnnotationKey] = strings.Join(x509Cert.Subject.StreetAddress, ",")
+		secret.Annotations[cmapi.SubjectPostalCodesAnnotationKey] = strings.Join(x509Cert.Subject.PostalCode, ",")
+		secret.Annotations[cmapi.SubjectSerialNumberAnnotationKey] = x509Cert.SerialNumber.String()
+>>>>>>> 10a9535a99ece4b45f2ea7d8c4aa57bd35d52768
 	}
 
 	return nil
 }
+
+// joinWithEscapeCSV returns the given list as a single line of CSV that
+// is escaped with quotes if necessary
+func joinWithEscapeCSV(in []string) (string, error) {
+	b := new(bytes.Buffer)
+	writer := csv.NewWriter(b)
+	writer.Write(in)
+	writer.Flush()
+
+	if err := writer.Error(); err != nil {
+		return "", fmt.Errorf("failed to write %q as CSV: %w", in, err)
+	}
+
+	s := b.String()
+
+	return s, nil
+}
```

Comment:
suggestion: as with the "reader" version of this function, this could use explicit tests I think.

Reply:
Thanks for your help and patience! I've added some additional tests that make use of these escaping functions which did help catch a small issue. Is there a good place to stick `joinWithEscapeCSV` and `splitWithEscapeCSV` so they can be used across the various places they are needed?

Reply:
(been on holiday, sorry for slow reply)

I'm not really precious about where they go; we've got so many `util` packages dotted around, I'm sure there's a home for the functions in one of those :stuck_out_tongue:

Reply:
Thanks @SgtCoDFish , indeed there are a lot. I will try and find a better home for it.